TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt

```diff
@@ -0,0 +1,106 @@
+package com.topsort.analytics.core
+
+import java.io.Closeable
+import java.io.IOException
+import java.io.OutputStream
+import java.net.HttpURLConnection
+import java.net.MalformedURLException
+import java.net.URL
+import java.util.zip.GZIPOutputStream
+
+const val LIBRARY_VERSION = 1.0
+
+data class HttpResponse(
+    val code : Int,
+    val message : String,
+){
+    fun isSuccessful() : Boolean {
+        return code in 200..299
+    }
+}
+
+class HttpClient (
+    private val apiHost: String,
+    private val requestFactory: RequestFactory = RequestFactory()
+) {
+
+    fun post(body: String, bearerToken: String?) : HttpResponse {
+        val connection: HttpURLConnection = requestFactory.upload(apiHost, bearerToken)
+        val postConnection = connection.createPostConnection()
+        val writeStream = postConnection.outputStream!!.bufferedWriter()
+
+        writeStream.write(body)
+        writeStream.flush()
+        postConnection.outputStream.close()
+
+        val response = HttpResponse(connection.responseCode, connection.responseMessage)
+        postConnection.close()
+
+        return response
+    }
+}
+
+/**
+ * Wraps an HTTP connection. Callers can either read from the connection via the [ ] or write to the connection via [OutputStream].
+ */
+abstract class Connection(
+    private val connection: HttpURLConnection,
+    val outputStream: OutputStream?
+) : Closeable {
+    @Throws(IOException::class)
+    override fun close() {
+        connection.disconnect()
+    }
+}
+
+internal fun HttpURLConnection.createPostConnection(): Connection {
+    val encoding = getRequestProperty("Content-Encoding") ?: ""
+    val outputStream: OutputStream =
+        if (encoding.contains("gzip")) {
```

Comment:
do we additionally wanna support deflate, br here? can the http library help us in any way?